### PR TITLE
FHIR Queries and QOL Updates

### PR DESCRIPTION
**Pull request by mgramigna**
# Summary

Various QOL updates to FHIR queries logic, plus a new test for single patient context. Also upgraded inferno core and Rakefile to adhere to correct version

I also removed uses of embedded cqf ruler. We don't need it. The scenario is designed to allow us to enter two different URLs. All uses of "embedded cqf ruler" across tests are replaced with another test input for the knowledge repository (confirmed with Bryn that this is a valid approach for testing data element submission scenario)

## New behavior

* `bundle exec rake db:migrate` works now to set up db locally (yay)
* FHIR Query tests are cool and work

## Code changes

* New test for FHIR queries where you can enter a patient ID and it replaces the `{{context.patientId}}` part of fhirQueryPattern validly (it's optional bc not all servers will support fhirQueryPattern)
* Update spec tests
* Remove uses of embedded cqf ruler (good riddance)
* Upgrade to inferno core 0.3 and fix Rakefile
* Update documentation to include proper running instructions (both docker and local)

# Testing guidance

``` bash
docker-compose pull
docker-compose up --build # can add --remove-orphans to remove an orphaned cqf-ruler service if you want
```

Run all tests. I recommend using `http://deqm_test_server:3000/4_0_1` as the data requirements knowledge repository server. For the FHIR queries stuff specifically, try out `https://cqf-ruler.ecqm.icfcloud.com/fhir` as the URL and see stuff actually work!

the second test will fail due to a bug in fqm execution, but the first one will pass

### CODE REVIEW

**Review by elsaperelli — APPROVED** [review]
Looks good to me!

**Review by sarahmcdougall — COMMENTED** [review]
Rspec tests all pass, and it's pretty cool to see the FHIR query logic actually work! 

One comment I have that I couldn't access from the code diff is: in the README under "Distributing Tests" we should change "The file in lib be what people use to import your gem..." to "The file in lib is what people use to import your gem..." Figured this change can be made now since we are making other README updates

File: spec/deqm_test_kit/submit_data_spec.rb [hunk 1]
```diff
@@ -112,12 +105,12 @@ def run(runnable, inputs = {})
         .to_return(status: 200, body: test_measure.to_json)
 
       # TODO: update, needs both get and post
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply spec/deqm_test_kit/submit_data_spec.rb]: Good catch, this can be removed

**Review by sarahmcdougall — APPROVED** [review]
:fire: :thumbsup: :fire:

**Pull request merged**